### PR TITLE
Es6 projects common modules email email

### PR DESCRIPTION
**Pull request by SiAdcock**
## What does this change?

It changes my perspective on life. Also Flow.

- Removed the singletons that were used to group loosely-related functions
- Removed the `opts` parameter from `updateForm` as it was not passed anywhere
- Effectively restricted the signature of `initEmail` to only accept iframe elements. Nowhere in the application is this function being called with a non-iframe element. Which leads on nicely to...
- Removes the `isIframed` parameter from `setup` as this can be inferred from the existence of the `iframeEl` parameter

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/email/email.js [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+import { trackNonClickInteraction } from 'common/modules/analytics/google';
+import { inlineSvg } from 'common/views/svgs';
+import successHtml from 'raw-loader!common/views/email/submissionResponse.html';
+import closeHtml from 'raw-loader!common/views/ui/close-button.html';
+import { getUserFromApi, isUserLoggedIn } from 'common/modules/identity/api';
+import userPrefs from 'common/modules/user-prefs';
+import uniq from 'lodash/arrays/uniq';
+
+import type { IdentityUser } from 'common/modules/identity/api';
+import type { bonzo } from 'bonzo';
```
Comment (sndrs) [inline static/src/javascripts/projects/common/modules/email/email.js]: do you need to import this?

Comment (SiAdcock) [reply static/src/javascripts/projects/common/modules/email/email.js]: Flow was giving me errors when I didn't

File: static/src/javascripts/projects/common/modules/email/email.js [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+import config from 'lib/config';
+import { getUserAgent } from 'lib/detect';
+import fetch from 'lib/fetch';
+import mediator from 'lib/mediator';
+import template from 'lodash/utilities/template';
+import { logError } from 'lib/robust';
+import { trackNonClickInteraction } from 'common/modules/analytics/google';
+import { inlineSvg } from 'common/views/svgs';
+import successHtml from 'raw-loader!common/views/email/submissionResponse.html';
+import closeHtml from 'raw-loader!common/views/ui/close-button.html';
```
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/email/email.js]: Bothe templates are only used by this module and are not longer than 5 lines. Could you convert them to inlined template literals?

File: static/src/javascripts/projects/common/modules/email/email.js [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+};
+
+const submitForm = (
+    $form: bonzo,
+    url: string,
+    analytics: Analytics
+): (Event => ?Promise<any>) => {
+    // simplistic email address validation to prevent misfired omniture events
+    const validate = (emailAddress: string): boolean =>
+        typeof emailAddress === 'string' && emailAddress.indexOf('@') > -1;
```
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/email/email.js]:
```js
emailAddress.includes('@')
```

**Review by sndrs — APPROVED** [review]
⭕️

**Pull request merged**